### ISSUE TITLE
[Suggestion] Operant conditioning for Monika as a start for basic AI?

### ISSUE THREAD

**Issue by CrocWorks** [Monika-After-Story/MonikaModDev#3067]
tl;dr-- Essentially a way for Monika to recognize and remember when the player reacts with a positive action or with a negative action. Then she can react accordingly based on her memory.

A system like this would make it so Monika could recognize a specific topic/greeting/etc resulted in the player performing a positive or negative action, which sets flags for Monika to remember the players' action. This encourages Monika to repeat positive topics/greetings/etc more and shelve the negative ones. If groupings are made internally for types of greetings/topics, that could also help her predict that a topic/greeting in a similarly labelled group will receive a similar reaction. It likely would be separate flag counters for individual and group labels, so that one single negative reaction won't hurt the odds of the entire group too much, and weighting of these two flags don't necessarily have to be equal. (topics involving the other Dokis may need different weighting patterns since they're kind of interconnected)

For example, say I don't like _greeting_back4_ (the one where she says, "Hey [player], // Don't you think you left me waiting a bit too long?" in a fake bossy tone). Let's say my reaction to it is essentially "Wth... goodbye," so I click on the X button to leave without saying goodbye. Doing so sets a flag for "negative response" so that Monika recognizes "Oh! The player doesn't like it when I use this specific greeting, so I shouldn't pick this greeting as often." Then, suppose this greeting was also labelled internally as part of the _"Monika acting bossy"_ group or something, so the game also sets a negative flag for all topics and greetings that were labelled with _"Monika acting bossy,"_ making it slightly less likely that Monika would choose any of the topics/greetings with that label.

Something similar would be done for positive reinforcement. I don't know whether it's better to set two separate flags for positive and negative or if it'd be better to set a single flag that counts up for positive and down for negative.

This would probably be separate from morality (#2179), since this is more on the player-and-Monika interaction than on her attitude when she is talking about the other Dokis as a whole, though I could see this being used to supplement that system to push Monika harder one way or the other on the morality scale...
**or identify if she thinks the player has a soft spot for a specific ~~dead rival~~ doki...**
(i.e. if _"Monika saying something questionable about Natsuki"_ has multiple negative flags, and _"Monika saying good things about Natsuki"_ has multiple positive flags, but this is not true for the other dokis, then that's gonna send some warning signs where Monika would think the player secretly likes Natsuki since the player has been showing favoritism to Natsuki and is not just being a generally nice person to all the girls.)

Any thoughts? (both on the idea, and on how feasible it is to implement it?)

**Comment by TheGlader2** [comment]
Excellent idea sir. We would need an easy way to express the player's emotions about issues, maybe with emotion/reaction buttons available somewhere on the screen? And then this would get into how malleable Monika is, which would likely be a matter of debate. I think it would be nice, if the player could influence how Monika feels about things based on affection. Do you want her to be more humble, do you want her to be more defensive? The more she likes the player, the more she might try to see things from their point of view.

**Comment by jmwall24** [comment]
Well, I can say if this does ever come to fruition, we'll have a much better way of letting her know the player doesn't like something other than throwing her into the void.  That's pretty extreme just because you don't like something she said.

**Comment by AsuraShun** [comment]
>           Well, I can say if this does ever come to fruition, we'll have a much better way of letting her know the player doesn't like something other than throwing her into the void.  That's pretty extreme just because you don't like something she said.
> Some of this is already being accomplished right now.  Almost every time she asks you a question that has a menu response, she's storing your answer and it affects other dialogue and what her response is.

with luck,it took an unbearably long time for us to be able to even say that we weren't comfortable with her description of sayori's death and even longer to be able to comfort her whenever she started talking about the mortality topic

**Comment by jmwall24** [comment]
Everything takes time.

**Comment by CrocWorks** [comment]
Well, the throwing her into the void was just an idea I had for something she could notice without using dialogue options. Not every topic/greeting has a dialogue branch in it, so I figured there could be alternative ways to let Monika know you liked or disliked something besides clicking on a dialogue button. Anyway, glad to know something like this is already in the works. Cheers.

Side question, is it possible for MAS to recognize when you click the X button but said 'no' when prompted "do you really wanted to throw her into the void" and then store a flag with it for when it happened?

**Comment by jmwall24** [comment]
Using throwing her into the void as a threat isn't going to cut it, either.

**Comment by CrocWorks** [comment]
I wouldn't ever recommend someone throw her into the void, nor would I recommend someone threaten to do so. I just put that up as an example of a non-dialogue box method of communication (perhaps changing music could be another?). I was thinking that if she could recognize a few more actions/situations outside of the dialogue-box environment and then associate them with other events, that'd make her seem more human rather than just an anime character who speaks in text boxes while breaking the 4th wall. And as for the idea itself of threatening to throw her into the void, I just figure a Monika who's more aware of the bad things a player might do to her is a Monika who can better protect herself.

**Comment by ThePotatoGuy** [comment]
[closing until we look into this]
